src/Switch-Keithley_707A/main.py

```diff
@@ -0,0 +1,145 @@
+# This Device Class is published under the terms of the MIT License.
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+#
+# Copyright (c) 2024 SweepMe! GmbH (sweep-me.net)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# SweepMe! driver
+# * Module: Switch
+# * Instrument: Keithley 707A
+
+from __future__ import annotations
+
+from EmptyDeviceClass import EmptyDevice
+
+
+class Device(EmptyDevice):
+    description = """
+        <p><strong>Usage:</strong></p>
+        <ul>
+        <li>Enter comma or semicolon separated configurations to close a crosspoint, e.g "A1,A4,C12" or "C3;D4".</li>
+        <li>Open means the relay is open/disconnected. No current will flow through. Closed means the relay is closed/
+        connected, and current will flow through.</li>
+        <li>All crosspoints will be opened before setting a new value.</li>
+
+        </ul>
+        <p>&nbsp;</p>
+        <p><strong>Known issues:</strong></p>
+        <p>In case SweepBox is used as Sweep value, comma-separated configurations can not be used at the moment as the 
+        comma is already used by the SweepBox to split the values to be set.</p>
+    """
+
+    def __init__(self) -> None:
+        """Set up device parameters."""
+        EmptyDevice.__init__(self)
+
+        self.shortname = "Keithley 707A"
+
+        self.port_manager = True
+        self.port_types = ["GPIB"]
+        self.port_properties = {
+            "timeout": 5,
+            "EOL": "\r",
+        }
+
+        self.variables = []
+        self.units = []
+        self.plottype = []
+        self.savetype = []
```

Comment:
What's the reason / difference between 707A/B why this one does not have any variables and the other drives has a channel variable? (Though channel does not really seem to be a channel but rather the Switching Matrix state?)

Reply:
The 707B returns self.value, which is a comma-separated string of all channels that are set to closed in apply. 
I was unsure if this information is of any use, as it is the same as the input value. To remain backward compatible, I did not change the return values of the 707B.
For 707A, we could return the input string as well (same problem, I do not think this is particular helpful) or we could query the channel state of the matrix and return the complete status as comma-separated string or any other data format. As these devices have typically >8x36 channels, the return value would be very large and the usefulness of this info is unclear.

Reply:
If we (later) add a sweep mode where the previous state of the matrix is not reset in apply, but rather updated, it could be helpful to return the updated matrix state for double-checking

Reply:
Return of self.value has been added to link the matrix state to the measurement result data.

Reply:
After adding self.value to the return value of call(), self.variables, ...units, etc. also need to be updated. Otherwise SweepMe! expects zero variables, but gets one value from call()